### PR TITLE
Fix cors origin error in the CLI

### PR DESCRIPTION
**Pull request by AbegaM**
## Modifications 

1. Added a validation in `/config/index.js` to fix a bug introduced in the previous pull request. The validation checks if `argv.cors` is falsy and ensures that the cors origin whitelist is set correctly based on the command line argument `-c` or the environment variable `CORS_ORIGIN_WHITELIST` only if they are defined:
    ```js
       cors: {
           origin: argv.cors?.split(',') ||
           envVars.CORS_ORIGIN_WHITELIST?.split(',') || ['*'],
        },
     ```

2. Modified the code in `src/index.js` to fix the issue with CORS not allowing access from all origins when the user passes `*` from the CLI:
     ```js
       const corsOrigin = config.cors.origin
       const corsOptions = {
           origin: corsOrigin[0] === '*' ? '*' : corsOrigin
        };
       app.use(cors(corsOptions))
      ```

### CODE REVIEW

**Comment by AbegaM** [comment]
Hello @thevahidal,

Could you please review this PR at your earliest convenience? When I submitted the last PR, there were some errors that prevented the CORS from allowing all origins when passing * to the -c flag. However, I have since fixed and tested everything. I would greatly appreciate it if you could review the changes I have made.

Thank you.

**Comment by thevahidal** [comment]
Hey @AbegaM, Sorry for the late response, I'll merge it asap.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Install Soul CLI with npm
 ## Usage
 
 Soul is command line tool, after installing it,
-Run `soul -d sqlite.db -p 8000` and it'll start a REST API on [http://localhost:8000](http://localhost:8000) and a Websocket server on [ws://localhost:8000](ws://localhost:8000).
+Run `soul -d sqlite.db -p 8000 -c "*"` and it'll start a REST API on [http://localhost:8000](http://localhost:8000) and a Websocket server on [ws://localhost:8000](ws://localhost:8000).
```
Comment (thevahidal) [inline README.md:21]: You provided `*` as the default cors option so I think it's better not to include it here

**Review by thevahidal — CHANGES_REQUESTED** [review]
Thanks for your PR, please also make sure to increase the minor version.

**Review by thevahidal — CHANGES_REQUESTED** [review]
Thanks for your PR, please also make sure to increase the minor version.

**Comment by AbegaM** [comment]
Hi @thevahidal 

I have fixed the issues, please review my changes when you have time and let me know if there is anything that needs to be modified.

**Comment by AbegaM** [comment]
Hi @thevahidal

I have pushed two commits and have fixed the issues, please review my changes when you have time and let me know if there is anything that needs to be modified.

**Review by thevahidal — APPROVED** [review]

**Comment by AbegaM** [comment]
@thevahidal, please let me know when you publish the changes to npm

**Pull request merged**